### PR TITLE
UCS/TYPE: Add a generic UCS_PARAM_VALUE macro

### PR DESCRIPTION
**Pull request by gleon99**
## What
Add a generic macro, which returns a param value from a params struct, after checking a corresponding condition flag in its `field_mask` - otherwise it returns a default value.

## Why ?
Avoid code repetition & duplication - have this logic in one single place instead.

### CODE REVIEW

**Comment by gleon99** [comment]
@yosefe  - I didn't add "set" macros, because I couldn't find any use case for them (an appropriate place in existing code).

File: src/uct/base/uct_iface.h [hunk 1]
```diff
@@ -124,8 +125,7 @@ enum {
 
 
 #define UCT_EP_PARAMS_GET_PATH_INDEX(_params) \
-    (((_params)->field_mask & UCT_EP_PARAM_FIELD_PATH_INDEX) ? \
-     (_params)->path_index : 0)
+    UCS_PARAM_VALUE(UCT_EP_PARAM_FIELD, _params, path_index, PATH_INDEX, 0)
```
Comment (yosefe) [inline src/uct/base/uct_iface.h:128]: maybe add UCT_EP_PARAM_VALUE?

File: src/uct/base/uct_iface.c [hunk 2]
```diff
@@ -161,17 +161,11 @@ void uct_iface_set_async_event_params(const uct_iface_params_t *params,
...
-    }
-
-    if (params->field_mask & UCT_IFACE_PARAM_FIELD_ASYNC_EVENT_ARG) {
-        *event_arg = params->async_event_arg;
-    } else {
-        *event_arg = NULL;
-    }
+    *event_cb  = UCT_IFACE_PARAM_VALUE(params, async_event_cb, ASYNC_EVENT_CB,
+                                       NULL);
+
```
Comment (yosefe) [inline src/uct/base/uct_iface.c:166]: minor: can remove extra spaces between UCT_IFACE_PARAM_VALUE statements (now that they are not if/else statements)

**Comment by yosefe** [comment]
@brminich can you pls review as well?

**Review by brminich — APPROVED** [review]

**Comment by gleon99** [comment]
bot:pipe:retest

File: src/ucp/core/ucp_context.c [hunk 3]
```diff
@@ -1290,50 +1290,33 @@ static ucs_status_t ucp_fill_resources(ucp_context_h context,
...
+    context->config.tag_sender_mask = UCP_PARAM_FIELD_VALUE(params,
+                                                            tag_sender_mask,
+                                                            TAG_SENDER_MASK, 0);
 
-    if (params->field_mask & UCP_PARAM_FIELD_REQUEST_SIZE) {
-        context->config.request.size = params->request_size;
-    } else {
-        context->config.request.size = 0;
-    }
+    context->config.request.size    = UCP_PARAM_FIELD_VALUE(params, request_size,
```
Comment (yosefe) [inline src/ucp/core/ucp_context.c:1303]: pls remove the extra spaces for all similar places in this PR

File: src/uct/base/uct_iface.h [hunk 4]
```diff
@@ -123,9 +124,13 @@ enum {
                     "UCT_EP_PARAM_FIELD_DEV_ADDR and UCT_EP_PARAM_FIELD_IFACE_ADDR are not defined")
 
 
+#define UCT_EP_PARAM_VALUE(_params, _name, _flag, _default) \
```
Comment (yosefe) [inline src/uct/base/uct_iface.h:127]: Are there additional places in UCT code that currently take ep params from field mask, and can be converted to use UCT_EP_PARAM_VALUE?

**Comment by gleon99** [comment]
@yosefe  - fixed.

**Review by yosefe — APPROVED** [review]

**Comment by gleon99** [comment]
@yosefe  - squashed.

**Comment by yosefe** [comment]
@gleon99 seems it has a conflict now

**Comment by gleon99** [comment]
@yosefe rebased over `master`.

**Review by yosefe — APPROVED** [review]

**Pull request merged**